src/battle_regulation.c

```diff
@@ -0,0 +1,276 @@
+#include "battle_regulation.h"
+
+#include "global.h"
+
+#include "constants/items.h"
+
+#include "dex_mon_measures.h"
+#include "party.h"
+#include "pokemon.h"
+#include "unk_020290B4.h"
+
+static BOOL sub_02074C50(u16 *a0, u16 *a1, u16 *a2, int a3, int a4, int a5, int a6);
+
+PokedexData *sub_02074944(HeapID heapId) {
+    PokedexData *pokedex = PokedexData_Create(heapId);
+    PokedexData_LoadAll(pokedex, 0, heapId);
+    return pokedex;
+}
+
+void sub_0207495C(PokedexData *pokedex) {
+    PokedexData_UnloadAll(pokedex);
+    PokedexData_Delete(pokedex);
+}
+
+BOOL sub_0207496C(LinkBattleRuleset *ruleset, Pokemon *pokemon, PokedexData *pokedex) {
+    u16 species = GetMonData(pokemon, MON_DATA_SPECIES, NULL);
+    if (ruleset == NULL) {
+        return TRUE;
+    }
+
+    int rule = LinkBattleRuleset_GetRuleValue(ruleset, 2);
```

Comment:
I'm guessing the one int is required to match? otherwise having descriptive variables like level would be more ideal

Reply:
Also you can change 2 to something like RULEVAL_LEVEL or whatever you deem appropriate, etc for all other known consts

Reply:
Handled in #332